src/Server.UI/Pages/Enrolments/Components/Location.razor

```diff
@@ -75,21 +77,37 @@
 
     protected override void OnParametersSet()
     {
-        if (Model is not null && Locations is not null && Locations.Any(l => l.Id == Model!.CurrentLocation.Id) == false)
+        if (Model is not null && Locations is not null)
         {
-            Model!.EnrolFromAlternativeLocation = true;
-            _mustEnrolInOtherLocation = true;
+            if (Locations.Any(l => l.Id == Model!.CurrentLocation.Id) == false)
+            {
+                Model!.EnrolFromAlternativeLocation = true;
+                _mustEnrolInOtherLocation = true;
+            }
+            else
+            if (Model!.AlternativeLocation != null &&  Model!.AlternativeLocation!.Id != Model!.CurrentLocation.Id)
+            {
+                Model.EnrolFromAlternativeLocation = true;
+                Model.AlternativeLocation = Locations.FirstOrDefault(x => x.Id == Model.AlternativeLocation.Id);                
+            }
```

Comment:
No need for null forgiving operator (`!`) in most instances here. Also, suggested using `is` and `is not` as opposed to `==` and `!=` as it reads a little easier.

```suggestion
            if (Locations.Any(l => l.Id == Model.CurrentLocation.Id) is false)
            {
                Model.EnrolFromAlternativeLocation = true;
                _mustEnrolInOtherLocation = true;
            }
            else if (Model.AlternativeLocation is not null && Model.AlternativeLocation.Id != Model.CurrentLocation.Id)
            {
                Model.EnrolFromAlternativeLocation = true;
                Model.AlternativeLocation = Locations.FirstOrDefault(x => x.Id == Model.AlternativeLocation.Id);                
            }
```